src/foam/dao/SQLData.java

```diff
@@ -0,0 +1,146 @@
+package foam.dao;
+
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+
+import java.util.*;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+
+/**
+ * prepares sql info based on a FObject
+ */
+public class SQLData {
+
+  //TODO(drish): use uuids ?
+  protected Object id;
+  protected String table;
+  protected ArrayList<String> columnNames;
+  protected ArrayList<Object> values;
+
+  public SQLData(FObject obj) throws IllegalStateException, IllegalAccessException {
+
+    table = obj.getClass().getSimpleName().toLowerCase();
```

Comment:
Using simplename will cause conflicts between DAOS for models in different packages with same name.